### PR TITLE
Add parameter names to synthesized record members

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/45621

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
                     Parameters: ImmutableArray.Create<ParameterSymbol>(
                                     new SourceSimpleParameterSymbol(owner: this,
                                                                     TypeWithAnnotations.Create(ContainingType, NullableAnnotation.Annotated),
-                                                                    ordinal: 0, RefKind.None, "", isDiscard: false, Locations)),
+                                                                    ordinal: 0, RefKind.None, "other", isDiscard: false, Locations)),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs:48]: >"other" [](start = 94, length = 7)

Consider if it makes sense to make a constant to share with BaseEquals

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs [hunk 2]
```diff
@@ -1130,7 +1130,7 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_MultipleRecordParameterLists, "(int X)").WithLocation(13, 17)
                 );
 
-            Assert.Equal(new[] { "C..ctor(System.Int32 X, System.Int32 Y)", "C..ctor(C )" }, comp.GetTypeByMetadataName("C")!.Constructors.Select(m => m.ToTestDisplayString()));
+            Assert.Equal(new[] { "C..ctor(System.Int32 X, System.Int32 Y)", "C..ctor(C original)" }, comp.GetTypeByMetadataName("C")!.Constructors.Select(m => m.ToTestDisplayString()));
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1133]: Would it make sense to include a test which asserts that a simple positional record has *no* members containing parameters with an empty string name?

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1133]: I think this scenario may be adequately addressed by existing tests which verify the symbol display of all members in the record. I am just thinking of a situation where additional synthesized members are introduced later, it is easiest to just name any parameters at the same time the new member is introduced.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1133]: I agree that we should name parameters if we add any.
I'm pretty sure scenario with no positional members is covered already. Thanks

**Pull request merged**